readme.md

```diff
@@ -177,10 +178,98 @@ class NoOpHTTPClient
   end
 end
 
-notifier = Slack::Notifier.new 'WEBHOOK_URL', http_client: NoOpHTTPClient
+notifier = Slack::Notifier.new 'WEBHOOK_URL' do
+  http_client NoOpHTTPClient
+end
+```
+
+
+## Middleware
+
+By default slack-notifier ships with middleware to format links in the message & text field of attachments. You can configure the middleware a notifier will us on initialization:
+
+```ruby
+notifier = Slack::Notifier.new "WEBHOOK_URL" do
+  middleware format_message: { formats: [:html] }
+end
+# this example will *only* use the format_message middleware and only format :html links
+
+notifier.post text: "Hello <a href='http://example.com'>world</a>! [visit this](http://example.com)"
+# => will post "Hello <http://example.com|world>! [visit this](http://example.com)"
+```
+
+The middleware can be set with a their name, or by name and options. They will be triggered in order.
+
+```ruby
+notifier = Slack::Notifier.new "WEBHOOK_URL" do
+  middleware :format_message, :format_attachements
+end
+# will run format_message then format_attachments with default options
+
+notifier = Slack::Notifier.new "WEBHOOK_URL" do
+  middleware format_message: { formats: [:html] },
+             format_attachements: { formats: [:markdown] }
+end
+# will run format_message w/ formats [:html] then format_attachements with formats [:markdown]
+```
+
+Available middleware:
+
+**`format_message`**
+
+This middleware takes the `:text` key of the payload and runs it through the [`Linkformatter`](#links). You can configure which link formats to look for with a `:formats` option. You can set `[:html]` (only html links), `[:markdown]` (only markdown links) or `[:html, :markdown]` (the default, will format both).
+
+**`format_attachments`**
+
+This middleware takes the `:text` key of any attachment and runs it through the [`Linkformatter`](#links). You can configure which link formats to look for with a `:formats` option. You can set `[:html]` (only html links), `[:markdown]` (only markdown links) or `[:html, :markdown]` (the default, will format both).
+
+
+### Writing your own Middleware
+
+Middleware is fairly straightforward, it is any class that inherits from `Slack::Notifier::PayloadMiddleware::FormatMessage::Base` and responds to `#call`. It will always be given the payload as a hash and should return the modified payload as a hash.
```

Comment:
I think you meant `Slack::Notifier::PayloadMiddleware::Base` instead of `Slack::Notifier::PayloadMiddleware::FormatMessage::Base`